tools/codegen/api/dispatcher.py

```diff
@@ -137,7 +137,22 @@ def cppargument_exprs(
     else:
         assert_never(a)
 
-def cpparguments_exprs(args: Sequence[CppArgumentPack]) -> Sequence[DispatcherExpr]:
+def cpparguments_exprs(func: FunctionSchema, * , method: bool, api_is_faithful: bool) -> Sequence[DispatcherExpr]:
+    dispatcher_calling_convention_is_faithful = local.use_c10_dispatcher().dispatcher_uses_new_style()
+    arguments = cpp.group_arguments(func, method=method, faithful=dispatcher_calling_convention_is_faithful)
+
+    if api_is_faithful:
+        argument_packs = tuple(
+            cpp.argument_faithful(a) for a in arguments
+        )
+    else:
+        argument_packs = tuple(
+            cpp.argument(a) for a in arguments
+        )
+
+    return _cpparguments_exprs(argument_packs)
+
+def _cpparguments_exprs(args: Sequence[CppArgumentPack]) -> Sequence[DispatcherExpr]:
```

Comment:
Not for this PR, but I hope you can see that CppArgumentPack is totally unnecessary now: the only difference is whether or not you construct a CppTensorOptionsArgumentPack or not, but we can just get rid of CppArgumentPack entirely and make sure we pass in faithful (which controls whether or not we constructed a CppTensorOptionsArgumentPack or not). This should make a number of things simpler.